diff -burp a/my_config.h b/my_config.h
--- a/my_config.h	2014-10-09 19:32:46.000000000 -0400
+++ b/my_config.h	2014-10-09 19:35:12.000000000 -0400
@@ -641,17 +641,4 @@
 #define SIZEOF_TIME_T 8
 /* #undef TIME_T_UNSIGNED */

-/*
-  stat structure (from <sys/stat.h>) is conditionally defined
-  to have different layout and size depending on the defined macros.
-  The correct macro is defined in my_config.h, which means it MUST be
-  included first (or at least before <features.h> - so, practically,
-  before including any system headers).
-
-  __GLIBC__ is defined in <features.h>
-*/
-#ifdef __GLIBC__
-#error <my_config.h> MUST be included first!
-#endif
-
 #endif

